aggsender/query/multisig_committee_query_test.go

```diff
@@ -82,9 +109,108 @@ func Test_ECDSAMultisigCommitteeQuery_GetMultisigCommittee(t *testing.T) {
 				require.NotNil(t, committee)
 				require.Equal(t, tc.threshold, committee.Threshold())
 				require.Len(t, committee.Signers(), tc.expectedNumSigners)
+				if tc.expectedSigner != "" {
+					require.Equal(t, tc.expectedSigner, committee.String())
+				}
 			}
 
 			mockSC.AssertExpectations(t)
 		})
 	}
 }
+
+func Test_CommitteeURLOverride(t *testing.T) {
+	t.Run("ReplaceURL replaces URLs based on the override map", func(t *testing.T) {
+		override := &CommitteeOverride{
+			URLMapping: map[string]string{
+				"http://original1": "http://override1",
+				"http://original3": "http://override3",
+			},
+		}
+
+		committee := []aggchainbase.IAggchainSignersSignerInfo{
+			{
+				Addr: common.HexToAddress("0x1"),
+				Url:  "http://original1",
+			},
+			{
+				Addr: common.HexToAddress("0x2"),
+				Url:  "http://original2",
+			},
+			{
+				Addr: common.HexToAddress("0x3"),
+				Url:  "http://original3",
+			},
+		}
+
+		expected := []aggchainbase.IAggchainSignersSignerInfo{
+			{
+				Addr: common.HexToAddress("0x1"),
+				Url:  "http://override1",
+			},
+			{
+				Addr: common.HexToAddress("0x2"),
+				Url:  "http://original2",
+			},
+			{
+				Addr: common.HexToAddress("0x3"),
+				Url:  "http://override3",
+			},
+		}
+
+		result := override.ReplaceURL(committee)
+		require.Equal(t, expected, result)
+	})
+	t.Run("ReplaceURL returns nil if override is nil", func(t *testing.T) {
+		var override *CommitteeOverride = nil
+
+		committee := []aggchainbase.IAggchainSignersSignerInfo{
+			{
+				Addr: common.HexToAddress("0x1"),
+				Url:  "http://original1",
+			},
+		}
+
+		result := override.ReplaceURL(committee)
+		require.Equal(t, committee, result)
+	})
+	t.Run("ReplaceURL returns nil if override map is empty", func(t *testing.T) {
+		override := &CommitteeOverride{
+			URLMapping: map[string]string{},
+		}
+
+		committee := []aggchainbase.IAggchainSignersSignerInfo{
+			{
+				Addr: common.HexToAddress("0x1"),
+				Url:  "http://original1",
+			},
+		}
+
+		result := override.ReplaceURL(committee)
+		require.Equal(t, committee, result)
+	})
```

Comment:
If it is easily doable to refactor it to a table test cases, it'd be nice to do 👍